### PR TITLE
RD-5760 Part 1 - Migrate files from containers directory and related components to TypeScript

### PR DESCRIPTION
**Pull request by qooban**
## Description

As a preparation for `app/components` directory restructure this PR covers TS migration of `app/containers` directory (only root files, the rest planned as part of another tickets).

In the next PR I'm planning to do actual restructuring.

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [ ] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
1. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3976/) (01.12.2022) - looks like only known issues, so I'm considering it as PASSED
2. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3999/) (05.12.2022) - looks like some known issues (RD-6526, RD-6522), but also something new
3. [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/4018/) (06.12.2022) - executed for very long time (RD-6549), so I stopped it
4. [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/4046/) (07.12.2022) - many tests failed (seems unrelated, but I don't feel confident to merge this PR)
5. [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/4074/) (09.12.2022) - comparing to `master` branch additional failures: `blueprints_spec.ts`,`deployments_spec.ts`,`deployments_view_spec.ts`
6. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4079/) (09.12.2022) - only specs not failing on `master` branch (`blueprints_spec.ts`,`deployments_spec.ts`,`deployments_view_spec.ts`) - only `deployments_view_spec.ts` failed this time
7. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4091/) (12.12.2022) - final verification after major UI System Test improvement ([RD-6544](https://github.com/cloudify-cosmo/cloudify-build-system/pull/822)) - 2 failures
    * `widgets/outputs_capabilities_spec` related to https://github.com/cloudify-cosmo/cloudify-stage/pull/2288, present on `master` branch as well
    * `templateManagement/pages_edit_spec` - might be related to my changes, so I improved this test robustness - https://github.com/cloudify-cosmo/cloudify-stage/pull/2292/commits/f04b64035865d762ddb6eafd37144e8282bf2a69
8. [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/4094/) (12.12.2022) - 3 failures
    * `widgets/outputs_capabilities_spec` - known issue (result of #2288 )
    * `flows/user_spec` - known issues (result of #2303 )
    * `widgets/deployments_spec` - new issue... But seems unrelated, looks like timeout/connectivity issue.

## Documentation
N/A

### CODE REVIEW

File: app/components/AddWidgetModal.tsx [hunk 1]
```diff
@@ -96,23 +106,39 @@ const StyledItem = styled(Item)`
 
 let nameIndex = 0;
 
-function generateCategories(widgets) {
-    const categories = widgets.reduce((curr, next) => {
+type Category = Stage.Types.ObjectKeys<typeof GenericConfigType['CATEGORY']>;
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/AddWidgetModal.tsx:109]: I think it requires a bit more effort as here I'm using `Stage` namespace from `StageAPI.ts`, so it's not possible to import `ObjectKeys` explicitly.

File: test/jest/components/LicensePage.test.tsx [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
 describe('(Component) LicensePage', () => {
-    let licensePageComponent = null;
-    let messageContainerComponent = null;
-    global.Stage = { Basic: BasicComponents };
-    global.moment = (timestamp, inputPattern) => moment(timestamp, inputPattern).utcOffset(0);
-
-    const verifyProps = (canUploadLicense, isProductOperational, license, status) => {
+    let licensePageComponent: ReactWrapper<LicensePageProps>;
+    let messageContainerComponent: ReactWrapper;
+    (global as any).Stage = { Basic: BasicComponents };
```
[the remark this answers is not in the document]
Comment (qooban) [reply test/jest/components/LicensePage.test.tsx:29]: Actually I found out that this line is not necessary as we already define `global.Stage` for tests in `test/jest/setupAfterEnv.ts`. Removed and updated other tests as well.

File: app/components/AddWidgetModal.tsx [hunk 3]
```diff
@@ -96,23 +106,39 @@ const StyledItem = styled(Item)`
...
+interface CategoryObject {
```
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/AddWidgetModal.tsx:110]: Done.

File: app/components/AddWidgetModal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (qooban) [reply app/components/AddWidgetModal.tsx]: Done.

**Review by kubama — APPROVED** [review]

**Review by kubama — APPROVED** [review]

**Pull request merged**